### ISSUE TITLE
Subparsers of subparsers : missing parsing exception ?

### ISSUE THREAD

**Issue by Altyrost** [p-ranav/argparse#261]
While working with the example of subparsers in the documentation, with `git`:

`some_exe submodule` is considered valid by the parser, even if the subparser cannot have any other argument than the subparser `update`.

Did I miss something here ?

**Comment by skrobinson** [comment]
If I understand correctly, you expect an error if `update` is missing after `submodule` in the parsed string.

Unless a missing argument (or subparser) is `required`, the parser will not report an error. Since subparsers do not have a `required` setting, they can never be missing.

But you can add the same logic to the git example with `is_subcommand_used`.

```cpp
  if (program.is_subcommand_used(submodule_command) &&
      !submodule_command.is_subcommand_used(submodule_update_command)) {
    std::cout << "submodule command called without update subcommand" << std::endl;
  }
```

**Comment by Altyrost** [comment]
Right. Shouldn't it be considered an error during the `parse_arg` ?

**Comment by skrobinson** [comment]
I don't believe it is always an error. From a UI view, some authors may prefer that `some_exe submodule -h` shows that the `update` subcommand exists. An example from git is the stash command which has list, show, drop, etc. subcommands.

But, others may want to say `submodule` without `update` is an error and they can express that condition with `is_subcommand_used`.
